packages/time-picker/test/dom/time-picker.test.js

```diff
@@ -0,0 +1,54 @@
+import { expect } from '@esm-bundle/chai';
+import { fixtureSync } from '@vaadin/testing-helpers';
+import '../../src/vaadin-time-picker.js';
+
+describe('vaadin-time-picker', () => {
+  let timePicker;
+
+  // Ignore generated attributes to prevent failures
+  // when running snapshot tests in a different order
+  const SNAPSHOT_CONFIG = {
+    ignoreAttributes: ['id', 'aria-describedby', 'aria-labelledby', 'for']
+  };
+
+  beforeEach(() => {
+    timePicker = fixtureSync('<vaadin-time-picker></vaadin-time-picker>');
+  });
+
+  describe('shadow', () => {
+    it('default', async () => {
+      await expect(timePicker).shadowDom.to.equalSnapshot();
+    });
+
+    it('disabled', async () => {
+      timePicker.disabled = true;
+      await expect(timePicker).shadowDom.to.equalSnapshot();
+    });
+
+    it('readonly', async () => {
+      timePicker.readonly = true;
+      await expect(timePicker).shadowDom.to.equalSnapshot();
+    });
+
+    it('invalid', async () => {
+      timePicker.invalid = true;
+      await expect(timePicker).shadowDom.to.equalSnapshot();
+    });
+
+    it('theme', async () => {
+      timePicker.setAttribute('theme', 'align-right');
+      await expect(timePicker).shadowDom.to.equalSnapshot();
+    });
```

Comment:
Would it make sense to test the `pattern` and `placeholder` properties as well?

Reply:
Done.

Reply:
This test can be dropped now:

https://github.com/vaadin/web-components/blob/07e698552184497c134d6d8f72f66845eb73afcf/packages/time-picker/test/time-picker.test.js#L180-L184

Reply:
And this one:

https://github.com/vaadin/web-components/blob/07e698552184497c134d6d8f72f66845eb73afcf/packages/time-picker/test/time-picker.test.js#L186-L190

Reply:
And this one:

https://github.com/vaadin/web-components/blob/07e698552184497c134d6d8f72f66845eb73afcf/packages/time-picker/test/time-picker.test.js#L198-L201

Reply:
Removed tests for `pattern` and `placeholder`. Regarding `disabled` and `readonly`, these were missing as snapshots only covered shadow DOM (propagating to `vaadin-input-container`). Added those as well.